frontend/lib/create_listing/view/create_listing.view.dart

```diff
@@ -0,0 +1,414 @@
+            ),
+            if (showError)
+              const Padding(
+                padding: EdgeInsets.only(top: 4),
+                child: Text(
+                  'Select a condition to continue.',
+                  style: TextStyle(
+                    color: Color(0xFFFF5252),
+                    fontSize: 12,
+                  ),
+                ),
+              ),
+          ],
+        );
+      },
+    );
+  }
+
+  Widget _buildConditionChip(String condition, CreateListingViewModel model) {
+    final isSelected = model.condition == condition;
+    return Padding(
+      padding: const EdgeInsets.only(right: 8),
+      child: ChoiceChip(
+        label: Text(
+          condition,
+          style: TextStyle(
+            color: isSelected ? Colors.black : Colors.white,
+          ),
+        ),
+        selected: isSelected,
+        selectedColor: Colors.grey[200],
+        backgroundColor: const Color(0xFF2C2C2C),
+        onSelected: (selected) {
+          if (selected) {
+            model.setCondition(condition);
+          }
+        },
+      ),
+    );
+  }
+
+  Widget _buildDescriptionField() {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        TextFormField(
+          controller: _descriptionController,
+          maxLines: null,
+          minLines: 5,
+          validator: (value) {
+            if (value != null) {
+              if (value.length > CreateListingModel.maxDescriptionLength) {
+                return 'Description must be less than ${CreateListingModel.maxDescriptionLength} characters';
+              }
+              if (value.isNotEmpty && value.trim().isEmpty) {
+                return 'Description cannot be only whitespace';
+              }
+            }
+            return null;
+          },
+          decoration: InputDecoration(
+            errorStyle: const TextStyle(
+              color: Colors.red,
+              fontSize: 12,
+              height: 0.8,
+            ),
+            hintText: 'Description (recommended)',
+            hintStyle: TextStyle(
+              color: Colors.grey[600],
+            ),
+            filled: true,
+            fillColor: Colors.grey[200],
+            border: OutlineInputBorder(
+              borderRadius: BorderRadius.circular(12),
+              borderSide: BorderSide.none,
+            ),
+            errorBorder: OutlineInputBorder(
+              borderRadius: BorderRadius.circular(12),
+              borderSide: const BorderSide(color: Colors.red, width: 1),
+            ),
+            focusedErrorBorder: OutlineInputBorder(
+              borderRadius: BorderRadius.circular(12),
+              borderSide: const BorderSide(color: Colors.red, width: 1),
+            ),
+            contentPadding: const EdgeInsets.symmetric(
+              horizontal: 16,
+              vertical: 14,
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+
+  void _submitForm() async {
```

Comment:
UI handlers like this, along with business logic should live in the ViewModel. The `setState` here should also likely be replaced with a `notifyListeners` call.